pkg/balancer-js/src/pool-utils/swapper.ts

```diff
@@ -0,0 +1,149 @@
+import { Signer } from '@ethersproject/abstract-signer';
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { AddressZero, Zero } from '@ethersproject/constants';
+import { ContractTransaction, PayableOverrides } from '@ethersproject/contracts';
+import { Provider } from '@ethersproject/providers';
+import invariant from 'tiny-invariant';
+import { Vault, Vault__factory, WETH, WETH__factory } from '@balancer-labs/typechain';
+import { BatchSwapStep, FundManagement, SingleSwap, SwapKind } from '../types';
+
+/**
+ * Converts a batch swap which interacts with only a single pool to an equivalent simple swap
+ * @param swap - a SingleSwap object describing the pool to perform the swap against.
+ * @returns an equivalent batch swap
+ */
+export const convertSingleToBatchSwap = (
+  swap: SingleSwap,
+  limit: BigNumberish
+): { kind: SwapKind; swaps: BatchSwapStep[]; assets: string[]; limits: BigNumber[] } => {
+  const assets = [swap.assetIn, swap.assetOut];
+  const swaps = [
+    {
+      poolId: swap.poolId,
+      assetInIndex: assets.indexOf(swap.assetIn),
+      assetOutIndex: assets.indexOf(swap.assetOut),
+      amount: swap.amount,
+      userData: swap.userData,
+    },
+  ];
+  const limits = swap.kind === SwapKind.GivenIn ? [BigNumber.from(limit), Zero] : [Zero, BigNumber.from(limit).mul(-1)];
```

Comment:
This will also be affected by the sorting.

Also, I think the limits are wrong. You made the order [in, out], and in 'given in' applied the limit to the first element (in). The a given in swap should have a limit on out, not in (i.e. on the amount calculated, not given).